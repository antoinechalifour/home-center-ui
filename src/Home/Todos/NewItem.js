import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { graphql } from 'react-apollo'
import gql from 'graphql-tag'

const Form = styled.form`
  display: flex;
  background: #fff;
  border-top: 1px solid rgba(0, 0, 0, .15);

  button {
    border: none;
    text-transform: uppercase;
    padding: 12px;
  }
`

const Input = styled.input`
  box-sizing: border-box;
  outline: none;
  display: block;
  width: 100%;
  flex: 1;
  font-size: inherit;
  font-family: inherit;
  border: none;
  background: none;
  padding: 12px 16px;
`

class NewItem extends Component {
  static ppropTypes = {
    listId: PropTypes.number.isRequired,
    mutate: PropTypes.func.isRequired
  }

  state = {
    value: ''
  }

  onChange = e => this.setState({ value: e.target.value })

  onSubmit = e => {
    e.preventDefault()

    this.props.mutate({
      variables: { listId: this.props.listId, text: this.state.value }
    })
    this.setState({ value: '' })
  }

  render () {
    return (
      <Form onSubmit={this.onSubmit}>
        <Input
          value={this.state.value}
          onChange={this.onChange}
          placeholder='What needs to be done?'
        />
        <button type='submit'>Add</button>
      </Form>
    )
  }
}

export default graphql(
  gql`
  mutation addListItem($listId: Int, $text: String) {
    addListItem(listId: $listId, text: $text) {
      id
    }
  }
`,
  {
    options: {
      refetchQueries: ['ListQuery']
    }
  }
)(NewItem)
