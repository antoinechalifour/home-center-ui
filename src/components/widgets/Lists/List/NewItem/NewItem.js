import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import MdAdd from 'react-icons/lib/md/add'

export default class NewItem extends Component {
  static ppropTypes = {
    addItem: PropTypes.func.isRequired
  }

  state = {
    value: ''
  }

  onChange = e => this.setState({ value: e.target.value })

  onSubmit = e => {
    e.preventDefault()

    this.props.addItem(this.state.value)
    this.setState({ value: '' })
  }

  render () {
    return (
      <Form onSubmit={this.onSubmit}>
        <Input
          value={this.state.value}
          onChange={this.onChange}
          placeholder='Add something interesting...'
        />
        <button type='submit'><MdAdd /></button>
      </Form>
    )
  }
}

const Form = styled.form`
  display: flex;

  button {
    border: none;
    text-transform: uppercase;
    padding: 12px;
    background: none;
    color: #fff;
    cursor: pointer;

    svg {
      font-size: 18px;
    }
  }
`

const Input = styled.input`
  box-sizing: border-box;
  outline: none;
  display: block;
  width: 100%;
  flex: 1;
  font-size: inherit;
  font-family: inherit;
  border: none;
  background: none;
  padding: 12px 16px;
  color: inherit;

  ::placeholder {
    color: #fff;
    opacity: .33;
  }
`
