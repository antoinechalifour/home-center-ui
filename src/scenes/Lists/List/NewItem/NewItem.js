import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

export default class NewItem extends Component {
  static ppropTypes = {
    addItem: PropTypes.func.isRequired
  }

  state = {
    value: ''
  }

  onChange = e => this.setState({ value: e.target.value })

  onSubmit = e => {
    e.preventDefault()

    this.props.addItem(this.state.value)
    this.setState({ value: '' })
  }

  render () {
    return (
      <Form onSubmit={this.onSubmit}>
        <Input
          value={this.state.value}
          onChange={this.onChange}
          placeholder='Add something interesting...'
        />
        <button type='submit'>Add</button>
      </Form>
    )
  }
}

const Form = styled.form`
  display: flex;
  background: rgba(0, 0, 0, .15);

  button {
    border: none;
    text-transform: uppercase;
    padding: 12px;
    background: transparent;
    color: #fff;
  }
`

const Input = styled.input`
  box-sizing: border-box;
  outline: none;
  display: block;
  width: 100%;
  flex: 1;
  font-size: inherit;
  font-family: inherit;
  border: none;
  background: none;
  padding: 12px 16px;
  color: inherit;

  :placeholder {
    color: #fff;
  }
`
