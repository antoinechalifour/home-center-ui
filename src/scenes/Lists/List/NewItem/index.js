import { graphql } from 'react-apollo'
import { addListItem } from 'queries/lists'
import NewItem from './NewItem'

const queryOptions = {}

export default graphql(addListItem, queryOptions)(NewItem)
