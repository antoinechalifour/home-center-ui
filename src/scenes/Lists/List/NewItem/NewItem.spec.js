import 'jest-styled-components'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import NewItem from './NewItem'

describe('scenes/Lists/List/NewItem', () => {
  it('Should submit the item when the button is clicked', () => {
    const props = {
      listId: 2,
      mutate: jest.fn()
    }
    const wrapper = shallow(<NewItem {...props} />)
    const preventDefault = jest.fn()

    wrapper.setState({ value: 'New item!' })
    wrapper.simulate('submit', { preventDefault })

    expect(props.mutate.mock.calls.length).toBe(1)
    expect(preventDefault.mock.calls.length).toBe(1)
    expect(props.mutate.mock.calls[0][0]).toEqual({
      variables: {
        listId: props.listId,
        text: 'New item!'
      }
    })
    expect(wrapper.state()).toEqual({ value: '' })
  })

  it('Should render correctly', () => {
    const props = {
      listId: 2,
      mutate: jest.fn()
    }

    const tree = renderer.create(<NewItem {...props} />)
    tree.getInstance().setState({ value: 'This item will be added' })

    expect(tree).toMatchSnapshot()
  })
})
