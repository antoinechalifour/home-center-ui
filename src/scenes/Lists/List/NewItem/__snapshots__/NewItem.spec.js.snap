// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scenes/Lists/List/NewItem Should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #fff;
  border-top: 1px solid rgba(0,0,0,.15);
}

.c0 button {
  border: none;
  text-transform: uppercase;
  padding: 12px;
}

.c1 {
  box-sizing: border-box;
  outline: none;
  display: block;
  width: 100%;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-size: inherit;
  font-family: inherit;
  border: none;
  background: none;
  padding: 12px 16px;
}

<form
  className="c0"
  onSubmit={[Function]}
>
  <input
    className="c1"
    onChange={[Function]}
    placeholder="What needs to be done?"
    value="This item will be added"
  />
  <button
    type="submit"
  >
    Add
  </button>
</form>
`;
